spec/controllers/works/importing_spec.rb

```diff
@@ -0,0 +1,106 @@
+          params = { urls: urls, importing_for_others: false, import_multiple: "works" }
+          get :import, params
+          expect(flash[:error]).to start_with "You cannot import more than #{max}"
+        end
+      end
+
+      context "the current user is an archivist" do
+        it "should error when importing over the maximum number of works" do
+          max = ArchiveConfig.IMPORT_MAX_WORKS_BY_ARCHIVIST
+          urls = Array.new(max + 1) { |i| "url#{i}" }.join(", ")
+          params = { urls: urls, importing_for_others: false, import_multiple: "works" }
+          allow_any_instance_of(User).to receive(:is_archivist?).and_return(true)
+
+          get :import, params
+          expect(flash[:error]).to start_with "You cannot import more than #{max}"
+
+          allow_any_instance_of(User).to receive(:is_archivist?).and_call_original
+        end
+
+        it "should error when importing over the maximum number of chapters" do
+          max = ArchiveConfig.IMPORT_MAX_CHAPTERS
+          urls = Array.new(max + 1) { |i| "url#{i}" }.join(", ")
+          params = { urls: urls, importing_for_others: false, import_multiple: "chapters" }
+          allow_any_instance_of(User).to receive(:is_archivist?).and_return(true)
+
+          get :import, params
+          expect(flash[:error]).to start_with "You cannot import more than #{max}"
+
+          allow_any_instance_of(User).to receive(:is_archivist?).and_call_original
+        end
+      end
+    end
+  end
+
+  # note: this non-action method renders a template before returning which Rspec can't process,
+  # so calling the method directly requires a rescue to swallow the Rspec error
+  describe "import_single" do
+    it "should display the correct error when a timeout occurs" do
+      allow_any_instance_of(StoryParser).to receive(:download_and_parse_story).and_raise(Timeout::Error)
+      expect(controller.send(:import_single, ["url1"], {})).to rescue # stop it whingeing about the response
+      expect(flash[:error]).to start_with "Import has timed out"
+      allow_any_instance_of(StoryParser).to receive(:download_and_parse_story).and_call_original
+    end
+
+    it "should display the correct error when a StoryParser error occurs" do
+      allow_any_instance_of(StoryParser).to receive(:download_and_parse_story).and_raise(StoryParser::Error.new("message"))
+      expect(controller.send(:import_single, ["url1"], {})).to rescue # stop it whingeing about the response
+      expect(flash[:error]).to start_with "We couldn't successfully import that work, sorry: message"
+      allow_any_instance_of(StoryParser).to receive(:download_and_parse_story).and_call_original
+    end
+  end
+
+  # note: this non-action method renders a template before returning which Rspec can't process,
+  # so calling the method directly requires a rescue to swallow the Rspec error
+  describe "import_multiple" do
+    it "should display the correct error when a timeout occurs" do
```

Comment:
Correct me if I'm wrong, but I don't think these are timeout errors, but rather url and other errors